protocol/rpcprovider/provider_load_manager.go

```diff
@@ -1,31 +1,49 @@
 package rpcprovider
 
 import (
+	"strconv"
 	"sync/atomic"
 )
 
 type ProviderLoadManager struct {
-	TotalSimultaneousRelays int64
-	ActiveRequestsPerSecond int64
+	rateLimitThreshold      atomic.Uint64
+	activeRequestsPerSecond atomic.Uint64
 }
 
-func (loadManager *ProviderLoadManager) addRelayCall() {
-	atomic.AddInt64(&loadManager.ActiveRequestsPerSecond, 1)
+func NewProviderLoadManager(rateLimitThreshold uint64) *ProviderLoadManager {
+	if rateLimitThreshold == 0 {
+		return nil
+	}
+	loadManager := &ProviderLoadManager{}
+
+	loadManager.rateLimitThreshold.Store(rateLimitThreshold)
+	loadManager.activeRequestsPerSecond.Store(0)
+
+	return loadManager
 }
 
-func (loadManager *ProviderLoadManager) removeRelayCall() {
-	atomic.AddInt64(&loadManager.ActiveRequestsPerSecond, -1)
+func (loadManager *ProviderLoadManager) addRelayCall() {
+	if loadManager == nil {
+		return
+	}
+	loadManager.activeRequestsPerSecond.Add(1)
 }
 
-func (loadManager *ProviderLoadManager) getRelayCallCount() int64 {
-	totalRelays := atomic.LoadInt64(&loadManager.ActiveRequestsPerSecond)
-	return totalRelays
+func (loadManager *ProviderLoadManager) subtractRelayCall() {
+	if loadManager == nil {
+		return
+	}
+	loadManager.activeRequestsPerSecond.Add(^uint64(0))
 }
 
-func (loadManager *ProviderLoadManager) getProviderLoad() float64 {
-	if loadManager.TotalSimultaneousRelays == 0 {
-		return 0
+func (loadManager *ProviderLoadManager) getProviderLoad() string {
+	if loadManager == nil {
+		return ""
+	}
+	loadedRateLimitThreshold := loadManager.rateLimitThreshold.Load()
+	if loadedRateLimitThreshold == 0 {
```

Comment:
we already know its not 0 otherwise loadManager is nil.

Reply:
Fixed in [56191f6](https://github.com/lavanet/lava/pull/1720/commits/56191f626d3c96ac8a2dc3ec7699922f6bf83e94)